numpy/ma/core.py

```diff
@@ -7085,7 +7122,12 @@ def power(a, b, third=None):
     # Get the masks
     ma = getmask(a)
     mb = getmask(b)
-    m = mask_or(ma, mb)
+    # m = mask_or(ma, mb)
+    """
+    Addressing issue 25589
+    Added shrink = False on line 7130 to preserve the existing mask.
```

Comment:
This comment is unnecessary to have in code. This is more of a git log message.

Reply:
Ok, thanks for letting me know.